### ISSUE TITLE
java.lang.illegalArguementException: Invalid format: ""

### ISSUE THREAD

**Issue by libBletchley** [SimpleMobileTools/Simple-Calendar#683]
A calendar can be created in Evolution 3.22.6 a couple different ways:
* webcal URL fed to Evolution
* manual entry
When a user creates a calendar by entering the data into Evolution manually, and then they export it as an `.ics` file, Simple Calendar pukes on the import of that file with this error:
```
java.lang.illegalArguementException: Invalid format: ""
```

When a webcal-sourced calendar is exported from Evolution and then imported into Simple Calendar, there is no issue.  So the first question is whether the bug is in Evolution or Simple Calendar.  I took two `.ics` files (one working, the other causing SC to crash) and supplied them to these ICS validation services:
* https://icalendar.org/validator.html
* http://ical-validator.herokuapp.com/validate/
Both files get a clean report from the first service, and from the 2nd service both get a the error
```
"! Specifying the charset in the MIME Content-Type is mandatory
Related RFC 5545 section: 3.1.4"
```
So I'm tempted to blame Simple Calendar.  Github blocks attaching `.ics` file types, so I've copied the contents of a failure case here:
```
BEGIN:VCALENDAR
CALSCALE:GREGORIAN
PRODID:-//Ximian//NONSGML Evolution Calendar//EN
VERSION:2.0
BEGIN:VTIMEZONE
TZID:/freeassociation.sourceforge.net/Europe/Brussels
X-LIC-LOCATION:Europe/Brussels
BEGIN:STANDARD
TZNAME:CET
DTSTART:19701028T030000
RRULE:FREQ=YEARLY;BYDAY=-1SU;BYMONTH=10
TZOFFSETFROM:+0200
TZOFFSETTO:+0100
END:STANDARD
BEGIN:DAYLIGHT
TZNAME:CEST
DTSTART:19700325T020000
RRULE:FREQ=YEARLY;BYDAY=-1SU;BYMONTH=3
TZOFFSETFROM:+0100
TZOFFSETTO:+0200
END:DAYLIGHT
END:VTIMEZONE
BEGIN:VEVENT
UID:20181029T134653Z-12192-1000-1-1@cypher
DTSTAMP:20181029T100322Z
DTSTART;TZID=/freeassociation.sourceforge.net/Europe/Brussels:
 20181108T090000
DTEND;TZID=/freeassociation.sourceforge.net/Europe/Brussels:
 20181108T093000
SEQUENCE:2
SUMMARY:bogus
LOCATION:yada
TRANSP:OPAQUE
DESCRIPTION:yada
CLASS:PUBLIC
CREATED:20181029T134714Z
LAST-MODIFIED:20181029T134714Z
END:VEVENT
END:VCALENDAR
```
Note that any whitespace-related issues may hide as a result of copy/paste.

**Comment by tibbi** [comment]
seems to work fine to me, might be affected by github whitespaces indeed. Please send the original .ics file to hello@simplemobiletools.com. Thanks

**Comment by libBletchley** [comment]
I have reproduced the problem in the android emulator.  So in principle you should be able to follow these steps to reproduce it:

*create a VM based on Android 5.0.1*
```
android create avd --name scdebug --target 'Google Inc.:Google APIs (x86 System Image):19' --device '5.1in WVGA'
```
*launch it*
```
emulator64-x86 -http-proxy https://localhost:8118 -no-boot-anim -debug-all -show-kernel -scale 0.9 -avd scdebug
```
*install Simple Calendar*
```
adb install com.simplemobiletools.calendar_127.apk
```
*make the root filesystem writable*
```
adb shell
su
mount -o rw,remount rootfs /
```
*copy the ICS file*
```
adb push bogus.ics
```
Launch Simple Calendar and import `/bogus.ics`.  Results in Java lang exception.. followed by a (false) "import successful" message.

**Comment by tibbi** [comment]
ah, didnt know you use an ancient version of the Calendar. Update it to 5.1.1 please and retry.

**Comment by libBletchley** [comment]
@tibbi 
> ah, didnt know you use an ancient version of the Calendar. Update it to 5.1.1 please and retry.

Makes no difference.  Both 5.1.1 and 5.1.2 have the same failure.

**Comment by tibbi** [comment]
try storing the file on the internal storage of your mobile, not at the root path. It could be something like /storage/emulated/0/... or so

**Comment by libBletchley** [comment]
@tibbi 

> try storing the file on the internal storage of your mobile, not at the root path. It could be something like /storage/emulated/0/... or so

I'm a bit baffled that import data sitting in a location where Simple Calendar has access would even be cause for suspicion.

On my unrooted phone I was in fact forced to stage the ICS file on the SD card (where there is user write permission).  On the emulated AVD I was forced to stage the file in `/` (because when pushing the file to `/storage/sdcard/` the file then appears only in adb-shell and not in the phone's interface).  For the AVD the root folder was apparently the only place that's mutually accessible from both `adb` and the apps.

Also note that external SD card storage is further proven to work with Simple Calendar running on a physical bare metal phone when the ICS file is generated from webcal data.

So either way it makes no difference where the file is accessed from w.r.t this bug.

Have you tried to reproduce this in the emulator?

**Comment by tibbi** [comment]
should be fixed in the pro version 6.0 released soon
